app/src/main/java/com/example/drawingApp/DrawingViewModel.kt

```diff
@@ -7,16 +7,85 @@ import androidx.lifecycle.ViewModel
 
 class DrawingViewModel : ViewModel() {
 
+    data class ViewState(
+        val color: Int,
+        val layers: MutableList<String>,
+        val activeBitmap: Bitmap?,
+        val isAlertDialogOpen: Boolean,
+        val layerDialogText: String,
+        val isLayerSheetOpen: Boolean,
+        val editIndex: Int?
+    )
 
     @ColorInt
     var primaryColor = Color.BLACK
 
-    val layerNames: MutableList<String> = mutableListOf()
+    private val layerNames: MutableList<String> = mutableListOf()
 
     var activeBitmap: Bitmap? = null
 
-    var isLayerCreationDialogOpen = false
-    var currentLayerCreationDialogText = ""
+    var isLayerCreationDialogOpen = false   //new Layer
+    var currentLayerDialogText = ""
 
+    var isViewingLayerSheetOpen = false       //edit layer
+
+    //var isLayerViewEditDialogOpen = false   //view inside edit layer
+    var layerViewEditDialogIndex: Int? = null
+    //var currentViewLayerDialogText = ""
+
+    var onUpdate: ((ViewState) -> Unit)? = null
+
+    private fun updateViewState() {
+        onUpdate?.invoke(
+            ViewState(
+                primaryColor,
+                layerNames,
+                activeBitmap,
+                isLayerCreationDialogOpen,
+                currentLayerDialogText,
+                isViewingLayerSheetOpen,
+                layerViewEditDialogIndex
+            )
+        )
+    }
+
+    fun initializeViews() {
```

Comment:
This is definitely initializing the layer, but is it really initializing a `view`? I could see this as just `initialize` but up to you

Reply:
renamed